lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm

```diff
@@ -418,23 +418,26 @@ sub update {
                         $testsuite_name = $spec;
                     }
 
-                    my $job_template_key
-                      = $arch . $product_name . $machine_name . $testsuite_name . ($job_template_name // '');
-                    die "Job template name '"
-                      . ($job_template_name // $testsuite_name)
-                      . "' is defined more than once. "
-                      . "Use a unique name and specify 'testsuite' to re-use test suites in multiple scenarios.\n"
-                      if $job_template_names{$job_template_key};
-                    $job_template_names{$job_template_key} = {
-                        prio              => $prio,
-                        machine_name      => $machine_name,
-                        arch              => $arch,
-                        product_name      => $product_name,
-                        product_spec      => $yaml_products->{$product_name},
-                        job_template_name => $job_template_name,
-                        testsuite_name    => $testsuite_name,
-                        settings          => $settings
-                    };
+                    $machine_names = [$machine_names] if ref($machine_names) ne 'ARRAY';
+                    foreach my $machine_name (@{$machine_names}) {
+                        my $job_template_key
+                          = $arch . $product_name . $machine_name . $testsuite_name . ($job_template_name // '');
+                        die "Job template name '"
+                          . ($job_template_name // $testsuite_name)
+                          . "' is defined more than once. "
+                          . "Use a unique name and specify 'testsuite' to re-use test suites in multiple scenarios.\n"
+                          if $job_template_names{$job_template_key};
+                        $job_template_names{$job_template_key} = {
+                            prio              => $prio,
+                            machine_name      => $machine_name,
+                            arch              => $arch,
+                            product_name      => $product_name,
+                            product_spec      => $yaml_products->{$product_name},
+                            job_template_name => $job_template_name,
+                            testsuite_name    => $testsuite_name,
+                            settings          => $settings
+                        };
```

Comment:
Now unfortunately there is one more level of indendation and IMHO way too many already. Since you and others managed to refactor in this regard recently I suggest to (at least) extract a method to reduce the obvious complexity of the current method.

Reply:
I guess this will look better after rebasing to master, as #2388 was merged.

Reply:
There's #2388 for that, no?